corehq/apps/hqcase/api/get_bulk.py

```diff
@@ -95,8 +95,10 @@ def _get_doc(doc_id):
         return _get_error_doc(doc_id)
 
     missing_ids = []
-    results_by_id = {res[es_id_field]: res for res in es_results if res['domain'] == domain}
-
+    results_by_id = {
+        res[es_id_field]: res for res in es_results
+        if res['domain'] == domain and res['type']
```

Comment:
> A case's case type should never be empty

That does sound right.

But the Case API v0.6 [does allow users](https://github.com/dimagi/commcare-hq/pull/33211) to create cases with an empty case type. Maybe we should change that. But until then I feel that the API should return all the cases that it allows user to create.

Reply:
You can also create cases with the type `""` via save-to-case (it can happen if your expression to evaluate the case type returns `""`, for instance).

I do think that the API should return all cases, even the weird ones, as it is a great tool for fixing bad data.  Say you have a bug in your form and make a bunch of real cases with missing case types.  One way to address that would be to use the API to find these cases and submit corrections ([you can change a case's type](https://github.com/dimagi/commcare-hq/blob/586c9a8e2b67d2d470233a1133f39080a3f64a76/corehq/apps/hqcase/tests/test_case_update_api.py#L186-L194)).  I also think that applies to deprecated case types - maybe there was a typo in an early version of your app and you need to migrate the existing data.

Reply:
Good point, I agree with the suggestion of removing the type check. This will bring it more in line with the list API, which currently returns `""` case types, unless the case type filter is used.